app/app.rb

```diff
@@ -0,0 +1,51 @@
+ENV['RACK_ENV'] ||= 'development'
+require 'sinatra/base'
+require 'sinatra/flash'
+require './app/models/data_mapper_setup'
+require './app/models/peep'
+require './app/models/user'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  set :session_secret, 'super secret'
+  register Sinatra::Flash
+
+  get '/posts' do
+    @peeps = Peep.all(:order => [:created_at.desc])
+    erb :index
+  end
+
+  get '/new' do
+    erb :new
+  end
+
+  post '/posts' do
+    Peep.first_or_create(post_text: params[:post_text], created_at: params[:created_at])
```

Comment:
I'd expect a `post` route to always create something (if the params are valid), rather than try to find something in the DB.

```ruby
# this
Peep.create(<params>)
# rather than this
Peep.first_or_create(<params>)
```